relayer/src/chain/cosmos.rs

```diff
@@ -523,15 +523,15 @@ impl CosmosSdkChain {
     fn account(&mut self) -> Result<&mut BaseAccount, Error> {
         if self.account == None {
             let account = self.block_on(query_account(self, self.key()?.account))?;
-
-            debug!(
-                sequence = %account.sequence,
-                number = %account.account_number,
-                "[{}] send_tx: retrieved account",
-                self.id()
-            );
-
-            self.account = Some(account);
+            if let Some(acc) = account.as_ref() {
+                debug!(
+                    sequence = %acc.sequence,
+                    number = %acc.account_number,
+                    "[{}] send_tx: retrieved account",
+                    self.id()
+                );
+            }
+            self.account = account;
```

Comment:
Thanks for the PR @tomtau!
`account` could be `None` here, but at line 540, we assume that the `account` has been cached. I would suggest that we return a `BaseAccount` from `query_account()` (just as we did before) and not `Option<BaseAccount>`, and either unwrap `EthAccount::decode().base_account` with a comment explaining why it is safe or use `ok_or()`.